src/Compilers/CSharp/Portable/BoundTree/BoundNode_Source.cs

```diff
@@ -0,0 +1,419 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal abstract partial class BoundNode
+    {
+#if DEBUG
+        /// <summary>
+        /// Gives an approximate printout of a bound node as C# code.
+        /// </summary>
+        internal string Source()
```

Comment:
Feel like this should be `GetSource` or `CalculateSource`. Given it's not a property expecting some verb.

Reply:
This is meant to be similar to `Dump()`. I'll change to `DumpSource()`.